### PR TITLE
Some general commits

### PR DESCRIPTION
**Pull request by tkaden4**

### CODE REVIEW

**Comment by Virepri** [comment]
== compares the pointers on strings for some reason. Don't ask me why.

Change that then I'll consider accepting the pull.

**Comment by Virepri** [comment]
Also, why remove the struct's parameters of data and type?

File: src/main.cpp [hunk 1]
```diff
@@ -18,80 +19,59 @@ int main(int argc, char* argv[])
...
-		}
-	}
-	Doesn't work and I'm too used to using C#.
-	*/
 
 	//Argument parsing
 	for(int i = 0;i < argc;i++)
 	{
 		std::string arg{argv[i]};
-		if(arg.compare("-o") == 0)
```
Comment (Virepri) [inline src/main.cpp]: == compares the pointer, .compare compares the data.

File: src/main.cpp [hunk 2]
```diff
@@ -18,80 +19,59 @@ int main(int argc, char* argv[])
...
+		if(arg == "-o")
 		{
 			outfile = argv[i+1];
 		}
-		else if(arg.compare("--out") == 0)
```
Comment (Virepri) [inline src/main.cpp]: == compares the pointer, .compare compares the data.

File: src/main.cpp [hunk 3]
```diff
@@ -18,80 +19,59 @@ int main(int argc, char* argv[])
...
+		else if(arg == "--out")
 		{
 			outfile = argv[i+1];
 		}
-		else if(arg.compare("-i") == 0)
```
Comment (Virepri) [inline src/main.cpp]: == compares the pointer, .compare compares the data.

File: src/main.cpp [hunk 4]
```diff
@@ -18,80 +19,59 @@ int main(int argc, char* argv[])
...
+		else if(arg == "-i")
 		{
 			infile = argv[i+1];
 		}
-		else if(arg.compare("--in") == 0)
```
Comment (Virepri) [inline src/main.cpp]: I think you get it by now.

File: src/scope/scope.h [hunk 5]
```diff
@@ -1,28 +1,21 @@
...
 
 #include <string>
 #include <vector>
 
-struct var 
-{
-	class c;
-	void *data;
-};
+struct var;
```
Comment (Virepri) [inline src/scope/scope.h]: Why this change?

File: src/scope/scope.cpp [hunk 6]
```diff
@@ -1,27 +1,17 @@
 #include "scope.h"
 
-scope::scope()
```
Comment (Virepri) [inline src/scope/scope.cpp]: Why remove the initialization code for the vectors & name?

File: src/scope/scope.cpp [hunk 7]
```diff
@@ -1,27 +1,17 @@
...
 
 scope::scope(std::string name_,std::vector<scope> subscopes_)
 {
 	name = name_;
 	subscopes = subscopes_;
-	variables = *new std::vector<var>();
 }
 
-scope::scope(std::string name_,std::vector<scope> subscopes_,std::vector<var> variables_)
+scope::scope(std::string name_,std::vector<scope> subscopes_, std::vector<var> variables_)
```
Comment (Virepri) [inline src/scope/scope.cpp]: oooooOOOOOOOOOOOOOOOOOHHHHHHHHH CHRIST IT'S **_A SINGLE SPACE THAT'S NOT EVEN CONSISTENT_**

Seriously, though, Why add a space onto this line, then not make it consistent for the previous statement?

**Comment by tkaden4** [comment]
== only compares pointers if you use them on pointer operands, such as (std::string *, or T *), but std::string, or any other object, is created on the stack by default, and as long as the == operator is overloaded within std::string, which it is, then we get a lexicographic comparison. I left the forward reference of "struct var;" to not break the declarations of data within the scope class. Otherwise, "class" is not a valid type.
